code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -597,134 +597,42 @@
 		return FALSE
 	if(!owner.Adjacent(A))
 		if(!silent)
-			to_chat(owner, span_warning("\The [A] is too far away."))
+			owner.balloon_alert(owner, "[A] is too far away")
 		return FALSE
-	if(isobj(A))
-		var/obj/O = A
-		if(CHECK_BITFIELD(O.resistance_flags, RESIST_ALL))
-			if(!silent)
-				to_chat(owner, span_warning("We cannot dissolve \the [O]."))
-			return FALSE
-		if(O.acid_check(acid_type))
-			if(!silent)
-				to_chat(owner, span_warning("This object is already subject to a more or equally powerful acid."))
-			return FALSE
-		if(istype(O, /obj/structure/window_frame))
-			var/obj/structure/window_frame/WF = O
-			if(WF.reinforced && acid_type != /obj/effect/xenomorph/acid/strong)
-				if(!silent)
-					to_chat(owner, span_warning("This [WF.name] is too tough to be melted by our weak acid."))
-				return FALSE
-	else if(isturf(A))
-		var/turf/T = A
-		if(T.acid_check(acid_type))
-			if(!silent)
-				to_chat(owner, span_warning("This object is already subject to a more or equally powerful acid."))
-			return FALSE
-		if(iswallturf(T))
-			var/turf/closed/wall/wall_target = T
-			if(wall_target.acided_hole)
-				if(!silent)
-					to_chat(owner, span_warning("[wall_target] is already weakened."))
-				return FALSE
-
-/obj/proc/acid_check(obj/effect/xenomorph/acid/new_acid)
-	if(!new_acid)
-		return TRUE
-	if(!current_acid)
-		return FALSE
-
-	if(initial(new_acid.acid_strength) > current_acid.acid_strength)
+	if(A.resistance_flags & UNACIDABLE || !(A.dissolvability(acid_type)))
```

Comment:
```suggestion
	if(A.resistance_flags & UNACIDABLE || A.dissolvability(acid_type) != 0)
```
Your code is correct yes, but it is more readable to show that dissolvability is and int